RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+        }
+
+        private string color;
+
+        public string Color
+        {
+            get { return color; }
+            set
+            {
+                if (!value.Equals(""))
+                    color = value;
+                else
+                    throw new ArithmeticException("Illegal color, color must be contain string");
+            }
+        }
+        private int numberOfShelves;
+
+        public int NumberOfShelves
+        {
+            get { return numberOfShelves; }
+            set
+            {
+                if (value >= 0)
+                    numberOfShelves = value;
+                else
+                    throw new ArithmeticException("Illegal numbershelves, numbershelves must be greater or equal zero");
+            }
+        }
+
+        public List<CShelf> shelfs;
+
+        public List<CShelf> Shelfs
+
+        {
+            get { return shelfs; }
+            //set
+            //{
+            //    if (!(value is null))
+            //    {
+            //        shelfs = value;
+            //    }
+            //    else
+            //    {
+            //        shelfs = new List<CShelf>();
+            //    }
+            //}
+        }
+
+        public CRefrigerator()
+        {
+            id = countObj++;
+            shelfs = new List<CShelf>();
+        }
+
+        public void AddShelfs(CShelf shelf)
+        {
+            if (shelfs.Count >= numberOfShelves)
+            {
+                Console.WriteLine("we cant add this shelfs, there is no place!!");
+            }
+            else
+            {
+                shelfs.Add(shelf);
+                Console.WriteLine("the shefs added!!");
+            }
+
+        }
+        public override string ToString()
+        {
+            StringBuilder result = new StringBuilder();
+            result.AppendLine($"Department ID: {Id}");
+            result.AppendLine($"Model: {Model}");
+            result.AppendLine($"Color: {Color}");
+            result.AppendLine($"Number of Shelves: {NumberOfShelves}");
+
+            // Print details for each shelf
+            result.AppendLine("Shelves:");
+
+            foreach (var shelf in Shelfs)
+            {
+                result.AppendLine(shelf.ToString());
+            }
+
+            return result.ToString();
+
+        }
+
+        //2
+        public double spaceLeftInRefrigerator()
+        {
+            double space = 0.0;
+            if (numberOfShelves >= 0)
+            {
+                for (int i = 0; i < shelfs.Count; i++)
+                {
+                    space = space + shelfs[i].spaceLeftInShelfs();
+                }
+            }
+            else
+            {
+                Console.WriteLine("Invalid number of shelves. Cannot calculate space.");
+            }
+
+            return space;
+        }
+
+        //3
+        public void pushToRefItem(CItem item1)
+        {
+
+            if (this.spaceLeftInRefrigerator() < item1.TakeSpace)
```

Comment:
This forces 2 iterations ob the fridge better to look for a shelf with space and if we did not find one print this error